tests/examples/runner.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+# Todo: Give this shell script a sane calling convention. Right now it has a
+# file containing the expected output as first argument and just executes
+# everything else.
+
+# Cleanup on errors / aborting
+cleanup() {
+    rm -rf "$file_stderr" "$file_stdout" || true
+}
+trap "cleanup" 0 2 3 15
```

Comment:
Put this after setting `file_stdout`/`file_stderr` to not call `rm -rf` with something empty.

Reply:
This is the same order as in `tests/run.sh` and calling `rm -rf` with an empty argument is harmless (it just fails). However, with the other way around, the same race will instead mean that the temporary file is not deleted even though it should.
Thus, I think it should stay the way it is.

Reply:
what about just standard `set -ueo pipefail` in the beginning of the shell script?
specifically `set -u` will guard from commands with unexpectedly blank args

Reply:
I'm not sure if that really is necessary here. Plus, so far this is a shell script and not a bash script, but this `set` assumes bash, I think.

Reply:
$ cat test.sh
set -u
A="test A"
echo $A $B

$ bash test.sh
test.sh: line 3: B: unbound variable

$ dash test.sh
test.sh: 3: test.sh: B: parameter not set

Reply:
Okay, so, should I amend the following into the last commit? I skipped pipefail since it is not in POSIX and the script does not use a single pipe (see http://pubs.opengroup.org/onlinepubs/009695399/utilities/set.html)
```diff
diff --git a/tests/examples/runner.sh b/tests/examples/runner.sh
index 8e5e3df5..ff209763 100755
--- a/tests/examples/runner.sh
+++ b/tests/examples/runner.sh
@@ -3,6 +3,8 @@
 # file containing the expected output as first argument and just executes
 # everything else.
 
+set -ue
+
 # Cleanup on errors / aborting
 cleanup() {
     rm -rf "$file_stderr" "$file_stdout" || true
@@ -16,8 +18,10 @@ expected_output=$1
 shift
 
 # Run the command that we were given
+set +e
 "$@" > ${file_stdout} 2> ${file_stderr}
 exit_code=$?
+set -e
 
 # If exit code is not zero or anything was produced on stderr...
 if [ $exit_code -ne 0 -o -s "${file_stderr}" ]
@@ -40,7 +44,7 @@ fi
 if ! cmp --silent "${file_stdout}" "${expected_output}"
 then
     echo "Expected text from ${expected_output}, but got:"
-    diff -u "${expected_output}" "${file_stdout}"
+    diff -u "${expected_output}" "${file_stdout}" || true
     exit 1
 fi
 
```
(Is there a nicer way to make `set -e` not break the actual execution of the target program?)

Reply:
change
```bash
 # Run the command that we were given
 set +e
 "$@" > ${file_stdout} 2> ${file_stderr}
 exit_code=$?
 set -e
```

to 

```
 # Run the command that we were given
 "$@" > ${file_stdout} 2> ${file_stderr} || exit_code=$?
```